### PR TITLE
Use a matrix of env variables and python versions

### PR DESCRIPTION
**Pull request by all3fox**
This allows to distribute CI load between more Travis machines.

Signed-off-by: Alexander Lisianoi <all3fox@gmail.com>

Refs: https://github.com/nexB/scancode-toolkit/issues/581

### CODE REVIEW

**Comment by pombredanne** [comment]
@all3fox Thanks. I need to review why the there is some test failure on Mac and Windows though

**Comment by pombredanne** [comment]
Is coveralls now able to merge several reports in one?

**Comment by all3fox** [comment]
You are right, it does not. Will update this PR to migrate to codecov then? You will only need to register for codecov and enable the repo. Will fix in the next 24 hours.

**Comment by pombredanne** [comment]
@all3fox yes, migrating to codecov is the way IMHO. I enabled it on this repo.

**Comment by all3fox** [comment]
Oh wow, sry for the coverage spam. An update:

1. Coveralls actually does report coverage from several jobs. Previously it did not work because of my configuration mistake.
2. Codecov does not quite work yet, I have no idea why and will take a look in the morning.

**Comment by all3fox** [comment]
Both coveralls and codecov work now:

* [coveralls report][1]
* [codecov report][2]

Also I tried to leave meaningful/helpful/explanatory comments directly in the `.travis.yml` while at the same time not being too wordy.

Appveyor build timed out but that is another issue. IMO this can be merged.

[1]: https://coveralls.io/builds/11666266
[2]: https://codecov.io/gh/nexb/scancode-toolkit/pull/613

**Comment by pombredanne** [comment]
@all3fox I guess it makes sense then to disable coveralls entirely before I merge?

**Comment by all3fox** [comment]
@pombredanne just deleting the `coveralls` from [here](https://github.com/all3fox/scancode-toolkit/blob/a5694911d20cc452e190b547c9d0aa3040b2e181/.travis.yml#L78) and [here](https://github.com/all3fox/scancode-toolkit/blob/a5694911d20cc452e190b547c9d0aa3040b2e181/.travis.yml#L89) should do the trick then. However, maybe just disable the coveralls bot?
